aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -213,6 +229,75 @@ static void isneginf_kernel_impl(TensorIterator& iter) {
   });
 }
 
+static void mode_kernel_impl(
+    Tensor& values,
+    Tensor& indices,
+    const Tensor& self,
+    int64_t dim,
+    bool keepdim) {
+  auto self_dim_size = ensure_nonempty_size(self, dim);
+  auto self_dim_stride = ensure_nonempty_stride(self, dim);
+
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "mode_cpu", [&] {
+    auto loop = [&](char** data, const int64_t* strides, int64_t n) {
+      auto* values_data_bytes = data[0];
+      auto* indices_data_bytes = data[1];
+      const auto* self_data_bytes = data[2];
+
+      std::vector<std::pair<scalar_t, int64_t>> elements(self_dim_size);
+      std::vector<int64_t> idxs(self_dim_size);
+      std::iota(idxs.begin(), idxs.end(), 0);
+
+      for (int64_t k = 0; k < n; ++k) {
+        scalar_t* values_data = (scalar_t*)values_data_bytes;
+        int64_t* indices_data = (int64_t*)indices_data_bytes;
+        const scalar_t* self_data = (scalar_t*)self_data_bytes;
+
+        scalar_t mode = 0;
+        int64_t modei = 0;
+        int64_t temp_freq = 0;
+        int64_t max_freq = 0;
+
+        std::transform(
+            idxs.begin(), idxs.end(), elements.begin(), [=](int64_t i) {
+                return std::make_pair(self_data[i * self_dim_stride], i);
+            });
```

Comment:
Am I right in saying `idxs` is never modified? In that case, I'd just make it a loop and not use transform. That way you don't need to allocate any vector.

Reply:
Yes, `idxs` won't change. But, won't `transform` just use the allocated space (instead of reallocating it)?

Reply:
Sure, it's not a huge performance issue. Just why allocate that memory at all?

Reply:
Ah, right. I think I wasn't getting exactly what you were talking about. That does make sense, though! I will do that.